src/vault/onchainLobVaultContract.ts

```diff
@@ -0,0 +1,198 @@
+    this.fastWaitTransactionInterval = options.fastWaitTransactionInterval ?? OnchainLobVaultContract.defaultFastWaitTransactionInterval;
+    this.fastWaitTransactionTimeout = options.fastWaitTransactionTimeout;
+
+    this.vaultContract = new Contract(options.vault.vaultAddress, lpManagerAbi, options.signer);
+  }
+
+  async approveTokens(params: ApproveVaultParams): Promise<ContractTransactionResponse> {
+    const token = this.vault.tokens.find(token => token.contractAddress === params.token);
+
+    if (!token) {
+      throw Error('Token is not in the pool.');
+    }
+    const tokenContract = new Contract(
+      token.contractAddress,
+      erc20Abi,
+      this.signer
+    );
+
+    const amount = this.convertTokensAmountToRawAmountIfNeeded(params.amount, token.decimals);
+    const tx = await this.processContractMethodCall(
+      tokenContract,
+      tokenContract.approve!(
+        params.vault,
+        amount,
+        {
+          gasLimit: params.gasLimit,
+          nonce: params.nonce,
+          maxFeePerGas: params.maxFeePerGas,
+          maxPriorityFeePerGas: params.maxPriorityFeePerGas,
+        }
+      ));
+
+    return tx;
+  }
+
+  async addLiquidity(params: AddLiquidityVaultParams): Promise<ContractTransactionResponse> {
+    const token = this.vault.tokens.find(token => token.contractAddress === params.token);
+
+    if (!token) {
+      throw Error('Token is not in the pool.');
+    }
+
+    const amount = this.convertTokensAmountToRawAmountIfNeeded(params.amount, token.decimals);
+    const amountUsd = this.convertTokensAmountToRawAmountIfNeeded(params.amountUsd, 18);
+    const minLpMinted = this.convertTokensAmountToRawAmountIfNeeded(params.minLpMinted, this.vault.lpToken.decimals);
+    const expires = getExpires();
+
+    const tx = await this.processContractMethodCall(
+      this.vaultContract,
+      this.vaultContract.addLiquidity!(
+        token.contractAddress,
+        amount,
+        amountUsd,
+        minLpMinted,
+        expires,
+        {
+          gasLimit: params.gasLimit,
+          nonce: params.nonce,
+          maxFeePerGas: params.maxFeePerGas,
+          maxPriorityFeePerGas: params.maxPriorityFeePerGas,
+        }
+      ));
+
+    return tx;
+  }
+
+  async removeLiquidity(params: RemoveLiquidityVaultParams): Promise<ContractTransactionResponse> {
+    const token = this.vault.tokens.find(token => token.contractAddress === params.token);
+
+    if (!token) {
+      throw Error('Token is not in the pool.');
+    }
+
+    const burnLP = this.convertTokensAmountToRawAmountIfNeeded(params.burnLP, this.vault.lpToken.decimals);
+    const minUsdValue = this.convertTokensAmountToRawAmountIfNeeded(params.minUsdValue, 18);
+    const minTokenGet = this.convertTokensAmountToRawAmountIfNeeded(params.minTokenGet, token.decimals);
+    const expires = getExpires();
+
+    const tx = await this.processContractMethodCall(
+      this.vaultContract,
+      this.vaultContract.addLiquidity!(
```

Comment:
typo

removeLiquidity
```suggestion
      this.vaultContract.removeLiquidity!(
```